test/suites/dev/test-precompile/test-precompile-referenda.ts

```diff
@@ -138,5 +140,225 @@ describeSuite({
+        const rawTxn = await context.writePrecompile!({
+          precompileName: "Referenda",
+          functionName: "submitAt",
+          args: [trackId, call.hash.toHex(), call.length, blockNumber.addn(1)],
+          rawTxOnly: true,
+        });
+
+        const block = await context.createBlock(rawTxn, {
+          expectEvents: [context.polkadotJs().events.referenda.Submitted],
+          signer: alith,
+        });
+
+        expectEVMResult(block!.result!.events, "Succeed");
+        const { data } = expectSubstrateEvent(block, "evm", "Log");
+
+        const evmLog: any = decodeEventLog({
+          abi: referendaAbi,
+          topics: data[0].topics.map((t) => t.toHex()) as [`0x${string}`],
+          data: data[0].data.toHex(),
+        });
+
+        expect(evmLog.eventName, "Wrong event").to.equal("SubmittedAt");
+        expect(evmLog.args.trackId, "Wrong event").to.equal(trackId);
+        expect(evmLog.args.hash, "Wrong event").to.equal(call.hash.toHex());
+      },
+    });
+
+    it({
+      id: "T05",
+      title: "should allow to submit after a certain block",
+      test: async function () {
+        const trackId = 0;
+        const call = context.polkadotJs().tx.identity.setIdentity({ display: { raw: "Me" } });
+        const blockNumber = await context.polkadotJs().query.system.number();
+        const rawTxn = await context.writePrecompile!({
+          precompileName: "Referenda",
+          functionName: "submitAfter",
+          args: [trackId, call.hash.toHex(), call.length, blockNumber.addn(1)],
+          rawTxOnly: true,
+        });
+
+        const block = await context.createBlock(rawTxn, {
+          expectEvents: [context.polkadotJs().events.referenda.Submitted],
+          signer: alith,
+        });
+
+        expectEVMResult(block!.result!.events, "Succeed");
+        const { data } = expectSubstrateEvent(block, "evm", "Log");
+
+        const evmLog: any = decodeEventLog({
+          abi: referendaAbi,
+          topics: data[0].topics.map((t) => t.toHex()) as [`0x${string}`],
+          data: data[0].data.toHex(),
+        });
+
+        expect(evmLog.eventName, "Wrong event").to.equal("SubmittedAfter");
+        expect(evmLog.args.trackId, "Wrong event").to.equal(trackId);
+        expect(evmLog.args.hash, "Wrong event").to.equal(call.hash.toHex());
+      },
+    });
+
+    it({
+      id: "T06",
+      title: "should allow to refund decision deposit",
+      test: async function () {
+        // Place deposit
+        const rawDepositTxn = await context.writePrecompile!({
+          precompileName: "Referenda",
+          functionName: "placeDecisionDeposit",
+          args: [proposalIndex],
+          rawTxOnly: true,
+        });
+
+        await context.createBlock(rawDepositTxn, {
+          expectEvents: [context.polkadotJs().events.referenda.DecisionDepositPlaced],
+          signer: alith,
+        });
+
+        // Cancel proposal
```

Comment:
We dont need comments like this since the function names already tell us the proposal is going to be cancelled

Reply:
You are right. On the other hand, my aim is to be consistent with regards to the comments. Place deposit -> Cancel proposal -> Refun deposit. When we remove the cancellation comment, the consistency deteriorates IMO; the comments and steps do not match. But anyway, I am removing it.